<launch>
    <!-- Add lidars to Warthog URDF -->
    <param name="robot_description" command="$(find xacro)/xacro $(find utexas_warthog_accessories)/urdf/lidar.urdf.xacro"/>

    <!-- Publish lidar static transform -->
    <node pkg="utexas_warthog_accessories" type="static_tf_publisher.py" name="static_tf_publisher" output="screen"/>

    <!-- Launch Ouster ROS driver -->
    <include file="$(find ouster_ros)/launch/driver.launch">
        <arg name="sensor_frame" value="front_lidar_link" />
        <arg name="sensor_hostname" value="192.168.131.128" />
        <arg name="timestamp_mode" value="TIME_FROM_ROS_TIME" />
        <arg name="viz" value="false" />
    </include>

    <!-- SLAM  -->
    <node pkg="pointcloud_to_laserscan" type="pointcloud_to_laserscan_node" name="pointcloud_to_laserscan">
        <remap from="cloud_in" to="ouster/points"/>
        <rosparam>
            min_height: -0.5 
            max_height: 0.25
            range_max: 100.0
        </rosparam>
    </node>
    <node pkg="gmapping" type="slam_gmapping" name="slam_gmapping" output="screen" />
</launch>